scripts/lib/file

```diff
@@ -46,24 +46,31 @@ spec:
 EOF
 }
 
-# create_operator_group_file creates an operator group for the operator CSV.
+# create_operator_group_file creates an operator group for the operator CSV if
+# the CSV does not support AllNamespaces.
 function create_operator_group_file() {
   local op_group_file="$1"
-  local name="$2"
-  local service_account="$3"
+  local op_name="$2"
+  local csv_file="$3"
   local namespace="$4"
 
-  cat <<EOF > "$op_group_file"
+  while read -r mode; do
+    if [[ "$(echo "$mode" | jq -r ".type")" == "AllNamespaces" && \
+      "$(echo "$mode" | jq -r ".supported")" == "false" ]]; then
+        local service_account="$(yq r "$csv_file" "spec.install.spec.deployments[0].spec.template.spec.serviceAccountName")"
+        cat <<EOF > "$op_group_file"
 apiVersion: operators.coreos.com/v1alpha2
 kind: OperatorGroup
 metadata:
-  name: ${name}-og
+  name: ${op_name}-og
   namespace: $namespace
 spec:
   serviceaccount: $service_account
```

Comment:
I'm not aware of OperatorGroup needing this field. I've used OperatorGroup without this field with no issues

Reply:
Its an optional field, I can remove it.